### PR TITLE
remove ssh-add flag and fix calling appendKnownHelper when not needed

### PR DESCRIPTION
**Pull request by ComradeProgrammer**
Related #16933 #15452
1. appendKnownHelper  should only be called when sshHost is true. appendKnownHelper should not be run when user is trying to connect to minikube's docker daemon with HTTP
2. before #15452 , to run minikube docker-env with ssh connection, user have to manually start a ssh-agent, and use ssh-add flag. But in #15452 , we added a feature, which automatically help users to do these two things, and user don't need to specify ssh-add or start a ssh-agent. However, no automation for "docker runtime with ssh connection". This makes docker-env's behavior inconsistent
3. so I think perhaps we should add this automation for  "docker runtime with ssh connection" too. And if so, we no longer need ssh-add flag. As long as ssh-host is set, we will automatically do all these things for the user(start ssh-agent, run ssh-add). Continuing to use ssh-add flag will make things  confusing.

@spowelljr  @medyagh

### CODE REVIEW

File: cmd/minikube/cmd/docker-env.go [hunk 1]
```diff
@@ -405,7 +405,7 @@ docker-cli install instructions: https://minikube.sigs.k8s.io/docs/tutorials/doc
 			exit.Error(reason.InternalDockerScript, "Error generating set output", err)
 		}
 
-		if sshAdd {
+		if sshHost {
```
Comment (spowelljr) [inline cmd/minikube/cmd/docker-env.go:408]: We may want to think about this as this is a breaking change. Scripts that were set to use `--ssh-add` will no longer execute this. We should leave the flag in and mark it deprecated in the flag description and tell users to use `--ssh-host` but have this be a `sshHost || sshAdd` to not break it.

File: cmd/minikube/cmd/docker-env.go [hunk 2]
```diff
@@ -324,9 +323,10 @@ docker-cli install instructions: https://minikube.sigs.k8s.io/docs/tutorials/doc
 			// docker-env on containerd depends on nerdctld (https://github.com/afbjorklund/nerdctld) as "docker" daeomn
 			// and nerdctld daemon must be used with ssh connection (it is set in kicbase image's Dockerfile)
 			// so directly set --ssh-host --ssh-add to true, even user didn't specify them
-			sshAdd = true
 			sshHost = true
+		}
 
+		if sshHost {
```
Comment (spowelljr) [inline cmd/minikube/cmd/docker-env.go:329]: Instead of having two `if sshHost` blocks, could we move this down to the second block?

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Review by medyagh — COMMENTED** [review]
lets keep things backward campatible

File: cmd/minikube/cmd/docker-env.go (hunk 2 above)
Comment (ComradeProgrammer) [reply cmd/minikube/cmd/docker-env.go:333]: Yes.
Now it has been done

File: cmd/minikube/cmd/docker-env.go [hunk 3]
```diff
@@ -326,7 +326,11 @@ docker-cli install instructions: https://minikube.sigs.k8s.io/docs/tutorials/doc
 			// so directly set --ssh-host --ssh-add to true, even user didn't specify them
 			sshAdd = true
 			sshHost = true
+		}
+
+		d := co.CP.Host.Driver
 
+		if sshHost {
```
Comment (medyagh) [inline cmd/minikube/cmd/docker-env.go:333]: this should be an OR so we dont break the old behavior

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/docker-env.go (hunk 3 above)
Comment (medyagh) [reply cmd/minikube/cmd/docker-env.go:333]: so the question is @afbjorklund would user ever need to use ssh-add without Docker using SSH-HOST ? 
so do we need to some reason keep the old behviour or ssh-add to be working regardless if the user is using SSH or HTTPS

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by medyagh** [comment]
lets go with most backward compatible approach

**Comment by ComradeProgrammer** [comment]
@medyagh  The comments have been solved. Maybe you can have a look at it now?

**Comment by medyagh** [comment]
@afbjorklund whats your opinion on this PR

File: cmd/minikube/cmd/docker-env.go [hunk 4]
```diff
@@ -125,7 +125,7 @@ type DockerShellConfig struct {
 var (
 	noProxy              bool
 	sshHost              bool
-	sshAdd               bool
+	sshAdd               bool // depecrated
```
Comment (afbjorklund) [inline cmd/minikube/cmd/docker-env.go:128]: deprecated

File: cmd/minikube/cmd/docker-env.go [hunk 5]
```diff
@@ -692,7 +693,7 @@ func init() {
 	defaultNoProxyGetter = &EnvNoProxyGetter{}
 	dockerEnvCmd.Flags().BoolVar(&noProxy, "no-proxy", false, "Add machine IP to NO_PROXY environment variable")
 	dockerEnvCmd.Flags().BoolVar(&sshHost, "ssh-host", false, "Use SSH connection instead of HTTPS (port 2376)")
-	dockerEnvCmd.Flags().BoolVar(&sshAdd, "ssh-add", false, "Add SSH identity key to SSH authentication agent")
+	dockerEnvCmd.Flags().BoolVar(&sshAdd, "ssh-add", false, "Depecrated: Add SSH identity key to SSH authentication agent")
```
Comment (afbjorklund) [inline cmd/minikube/cmd/docker-env.go:696]: Deprecated

**Review by afbjorklund — CHANGES_REQUESTED** [review]
Some typos.

Not sure it actually has to be deprecated, if it is truly not needed anymore than maybe it should just be removed from the documentation and made it into a no-op instead.

**Comment by afbjorklund** [comment]
But I never understood why we have to start a second `ssh-agent` anyway, usually it is started when you log in.

* #15452

File: site/content/en/docs/commands/docker-env.md [hunk 6]
```diff
@@ -28,7 +28,7 @@ minikube docker-env [flags]
       --no-proxy        Add machine IP to NO_PROXY environment variable
   -o, --output string   One of 'text', 'yaml' or 'json'.
       --shell string    Force environment to be configured for a specified shell: [fish, cmd, powershell, tcsh, bash, zsh], default is auto-detect
-      --ssh-add         Add SSH identity key to SSH authentication agent
+      --ssh-add         Depecrated: Add SSH identity key to SSH authentication agent
```
Comment (medyagh) [inline site/content/en/docs/commands/docker-env.md:31]: fix spelling

**Review by medyagh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**